.github/workflows/scripts/detect-impacted-packages.mjs

```diff
@@ -0,0 +1,291 @@
+      requiresMongo: packages.get(name).requiresMongo !== false,
+      requiresRedis: packages.get(name).requiresRedis === true
+    }))
+  };
+
+  /** @type {ImpactResult} */
+  const result = {
+    matrix,
+    hasPackages: matrix.include.length > 0
+  };
+
+  process.stdout.write(JSON.stringify(result));
+}
+
+async function loadPackages() {
+  const map = new Map();
+  let entries;
+  try {
+    entries = await fs.readdir(packagesRoot, { withFileTypes: true });
+  } catch (error) {
+    throw new Error(`Unable to read packages directory: ${error.message}`);
+  }
+
+  await Promise.all(entries
+    .filter((entry) => entry.isDirectory())
+    .map(async (entry) => {
+      if (IGNORED_PACKAGE_DIRS.has(entry.name)) {
+        return;
+      }
+      const packageDir = path.join(packagesRoot, entry.name);
+      const manifestPath = path.join(packageDir, 'package.json');
+      let manifest;
+      try {
+        const raw = await fs.readFile(manifestPath, 'utf8');
+        manifest = JSON.parse(raw);
+      } catch (error) {
+        console.warn(`Skipping ${entry.name}: ${error.message}`);
+        return;
+      }
+
+      if (!manifest.name) {
+        console.warn(`Skipping ${entry.name}: missing package name`);
+        return;
+      }
+
+      const scripts = manifest.scripts || {};
+      const hasTestScript = Boolean(scripts.test);
+
+      const dependencies = new Set([
+        ...Object.keys(manifest.dependencies || {}),
+        ...Object.keys(manifest.devDependencies || {}),
+        ...Object.keys(manifest.peerDependencies || {}),
+        ...Object.keys(manifest.optionalDependencies || {})
+      ]);
+
+      const testConfig = manifest.apostropheTestConfig || {};
+      const requiresMongo = testConfig.requiresMongo !== false;
+      const requiresRedis = testConfig.requiresRedis === true;
+
+      map.set(manifest.name, {
+        name: manifest.name,
+        relativeDir: path.posix.join('packages', entry.name),
+        dependencies,
+        hasTestScript,
+        requiresMongo,
+        requiresRedis
+      });
+    }));
+
+  return map;
+}
+
+function buildDirLookup(packages) {
+  const dirMap = new Map();
+  for (const pkg of packages.values()) {
+    dirMap.set(pkg.relativeDir, pkg.name);
+  }
+  return dirMap;
+}
+
+function resolveBaseSha() {
+  try {
+    const mergeBase = execSync(`git merge-base HEAD origin/${defaultBranch}`, {
+      encoding: 'utf8',
+      stdio: ['ignore', 'pipe', 'ignore']
+    }).trim();
+    return mergeBase || '';
+  } catch (error) {
+    return '';
+  }
+}
+
+function checkForceCI(baseSha) {
+  if (!baseSha) {
+    return false;
+  }
+
+  try {
+    const output = execSync(`git log ${baseSha}..HEAD --pretty=format:%s`, {
+      encoding: 'utf8',
+      stdio: ['ignore', 'pipe', 'ignore']
+    }).trim();
+    
+    if (!output) {
+      return false;
+    }
+    
+    const commitMessages = output.split('\n');
+    return commitMessages.some((msg) => /\[force ci\]/i.test(msg));
```

Comment:
So any commit on your PR containing this force ci will make them all run all the time (for this PR).

Reply:
It's just a quick way to solve  "not forget to run all tests for my change for some strange reason" . This follows the convention for github owned `(skip ci)`